### DISCUSSION TITLE
Running as a child process from NodeJS

### DISCUSSION THREAD

**Discussion by denvolok** [projectdiscovery/httpx#779] [Q&A] [answered]
### httpx version:
1.2.4
### Current Behavior:
I'm trying to run **httpx** from a NodeJS script(typescript):
```typescript
const { spawn } = require("node:child_process");

const cmd = spawn("httpx", [
  "-l",
  "/tmp/subdomains.txt",
  "-ct",
  "-v",
  "-debug",
]);

cmd.stdout.on("data", (data: any) => {
  console.log(`stdout: ${data}`);
});

cmd.stderr.on("data", (data: any) => {
  console.error(`stderr: ${data}`);
});

cmd.on("close", (code: any) => {
  console.log(`child process exited with code ${code}`);
});
```
I'm getting the following output, then the execution just stuck(I suppose before making any requests):
```typescript
    __    __  __       _  __
   / /_  / /_/ /_____ | |/ /
  / __ \/ __/ __/ __ \|   /
 / / / / /_/ /_/ /_/ /   |
/_/ /_/\__/\__/ .___/_/|_|
             /_/              v1.2.4

                projectdiscovery.io

Use with caution. You are responsible for your actions.
Developers assume no liability and are not responsible for any misuse or damage.


```
### Expected Behavior:
I expect to see the same output as if running command from cli.

### Steps To Reproduce:
1. `npx ts-node index.ts`

### Anything else:
**NodeJS version**: 16.17.0
Tested with Node's **exec** - the same problem.
Tested with native js - the same problem.
Tested **subfinder** - works as expected.

**Comment by forgedhallpass** [comment, marked as answer]
Try closing the `stdin` from your code:

```js
child.stdin.end()
```

![image](https://user-images.githubusercontent.com/13679401/190898758-9337e9f4-1891-46e6-96b6-62aa97389629.png)

**Comment by denvolok** [comment]
@forgedhallpass thanks, it works!
